Fortran/gfortran/utils/update-test-config.py

```diff
@@ -0,0 +1,633 @@
+    # for complex expressions, it can be used with simple expressions as well.
+    if t.startswith('{') and t.endswith('}'):
+        t = t[1:-1].strip()
+
+    # A simple expression may be a sequence of targets.
+    for tgt in t.split(' '):
+        if re_platform.match(tgt):
+            targets.append(convert_target_regex(tgt))
+        elif tgt in platforms:
+            targets.append(convert_target_regex(platforms[tgt]))
+        # Some "targets" need to be translated to compiler/linker flags.
+        elif tgt in ['fopenmp', 'fopenacc', 'pthread']:
+            options.append('-' + tgt)
+        elif tgt in ['c99_runtime']:
+            options.append('-lc')
+        elif tgt in [
+            'fd_truncate',
+            'fortran_large_int',
+            'fortran_real_10',
+            'fortran_real_16'
+        ]:
+            # FIXME: These may need something sane to be done.
+            pass
+        elif tgt in [
+            'arm_eabi',
+            'avx_runtime',
+            'fpic',
+            'libatomic_available',
+            'vect_simd_clones'
+        ]:
+            # As far as I can tell, nothing needs to be done for these targets.
+            pass
+        else:
+            warning('Unknown target: {}', tgt)
+
+    return targets, options
+
+# Parse the disabled targets from a target specification string.
+def parse_disabled_targets(t: str) -> list[str]:
+    targets = []
+
+    # An expression can be wrapped with braces. While this seems to be necessary
+    # for complex expressions, it can be used with simple expressions as well.
+    if t.startswith('{') and t.endswith('}'):
+        t = t[1:-1].strip()
+
+    # A simple expression may be a sequence of targets.
+    for tgt in t.split(' '):
+        if re_platform.match(tgt):
+            targets.append(convert_target_regex(tgt))
+        elif tgt in platforms:
+            targets.append(convert_target_regex(platforms[tgt]))
+        elif tgt in ['newlib']:
+            # FIXME: These may need something sane to be done.
+            pass
+        else:
+            warning('Unknown target to disable: {}', tgt)
+
+    return targets
+
+# Parse the target specification, if possible.
+# This is not guaranteed to parse all target specifications. We don't care
+# about the arbitrarily complex expressions that seem to be possible, so this
+# will only deal with "simple" expressions. Some of the target expressions
+# will be translated to compiler/linker flags. In those cases, update the
+# list of flags that are passed in.
+def parse_targets_into(
+    t: str, enabled_on: list[str], disabled_on: list[str], options: list[str]
+) -> None:
+    t = t.strip()
+
+    # An expression can be wrapped with braces. While this seems to be necessary
+    # for complex expressions, it can be used with simple expressions as well.
+    if t.startswith('{') and t.endswith('}'):
+        t = t[1:-1].strip()
```

Comment:
This could be moved to a `strip_braces` helper function, since it's used in 3 functions.